<launch>
    <arg name="debug" default="false"/>
    <set_env if="$(var debug)" name="ROSCONSOLE_CONFIG_FILE" value="$(find-pkg-share dsol)/launch/debug.conf"/>

    <arg name="tbb" default="0"/>
    <arg name="log" default="5"/>
    <arg name="vis" default="1"/>

    <arg name="camera" default="/camera"/>
    <arg name="use_depth" default="false"/>

    <node pkg="dsol" exec="sv_dsol_node_odom" name="dsol_odom" output="screen" namespace="dsol_odom"> # launch-prefix="xterm -e gdb -ex run --args">
        <param from="$(find-pkg-share dsol)/config/dsol.yaml"/>
        <param from="$(find-pkg-share dsol)/config/dsol_d455.yaml"/>

        <param name="tbb" value="$(var tbb)"/>
        <param name="log" value="$(var log)"/>
        <param name="vis" value="$(var vis)"/>
        <param name="use_depth" value="$(var use_depth)"/>

        <remap from="/dsol_odom/cinfo0" to="$(var camera)/infra1/camera_info"/>
        <remap from="/dsol_odom/cinfo1" to="$(var camera)/infra2/camera_info"/>
        <remap from="/dsol_odom/image0" to="$(var camera)/infra1/image_rect_raw"/>
        <remap from="/dsol_odom/image1" to="$(var camera)/infra2/image_rect_raw"/>
        <remap from="/dsol_odom/depth0" to="$(var camera)/depth/image_rect_raw"/>
        <remap from="/dsol_odom/gyr" to="$(var camera)/gyro/sample"/>
    </node>
</launch>